### PR TITLE
ATLAN-2570 fixed the hasMoreVertices condition for terminating graph correctly

### PR DESCRIPTION
**Pull request by suraj5077**
## Bug fix: https://linear.app/atlanproduct/issue/ATLAN-2570/find-hotel-20-or-[snowflake-lineage]-lineage-graph-not-showing-all

> Fixed the if condition and also removed redundant code.

## Type of change
- [x] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)

### Development

- [x] Lint rules pass locally
- [x] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### Code review 

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [x] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [x] Pull request linked to task tracker where applicable

### CODE REVIEW

**Comment by suraj5077** [comment]
@BurakAltas test cases here: https://www.notion.so/atlanhq/ATLAN-2570-test-cases-0fee3c7a7bcc435cb641b66de44b7efa
Please review

**Review by BurakAltas — APPROVED** [review]

**Comment by louisnow** [comment]
Can you add me to those test cases as well please or make it viewable by everyone in the company @suraj5077 ?
https://www.notion.so/atlanhq/ATLAN-2570-test-cases-0fee3c7a7bcc435cb641b66de44b7efa

**Comment by suraj5077** [comment]
Access shared @louisnow

**Comment by suraj5077** [comment]
@BurakAltas I have updated the PR and the test cases to fix the second issue we discovered. Team, please review

File: repository/src/main/java/org/apache/atlas/discovery/EntityLineageService.java [hunk 1]
```diff
@@ -467,47 +467,67 @@ private void addPaginatedVerticesToResult(boolean isInput,
...
+                currentOffset -= edgesOfProcess.size();
+        }
+    }
 
-                }
-                currentOffset = Math.max(0, currentOffset - 1);
+    private List<AtlasEdge> getEdgesOfProcess(boolean isInput, AtlasLineageContext lineageContext, Set<String> paginationCalculatedVertices, AtlasVertex processVertex) {
+        List<Pair<AtlasEdge, String>> processEdgeOutputVertexIdPairs = getUnvisitedProcessEdgesWithOutputVertexIds(isInput, lineageContext, paginationCalculatedVertices, processVertex);
+        processEdgeOutputVertexIdPairs.forEach(pair -> paginationCalculatedVertices.add(pair.getRight()));
+        List<AtlasEdge> edgesOfProcess = processEdgeOutputVertexIdPairs
```
Comment (BurakAltas) [inline repository/src/main/java/org/apache/atlas/discovery/EntityLineageService.java:496]: redundant assign, can be directly returned

File: repository/src/main/java/org/apache/atlas/discovery/EntityLineageService.java [hunk 2]
```diff
@@ -467,47 +467,67 @@ private void addPaginatedVerticesToResult(boolean isInput,
...
-            List<AtlasEdge> edgesOfProcess = processEdgeOutputVertexIdPairs
-                    .stream()
-                    .map(Pair::getLeft)
-                    .collect(Collectors.toList());
+            List<AtlasEdge> edgesOfProcess = getEdgesOfProcess(isInput, lineageContext, paginationCalculatedVertices, processVertex);
 
+            if (isFirstValidProcessReached)
+                currentOffset = 0;
             if (edgesOfProcess.size() > currentOffset) {
+                isFirstValidProcessReached = true;
```
Comment (BurakAltas) [inline repository/src/main/java/org/apache/atlas/discovery/EntityLineageService.java:483]: I was thinking like a solution that includes a separate `withinProcessOffset` which is responsible for the inner process offset. Then, at the end of the iteration (depending on the limit), merging it into the `currentOffset`

Comment (suraj5077) [reply repository/src/main/java/org/apache/atlas/discovery/EntityLineageService.java:483]: This seemed like an easier to understand logic @BurakAltas , just identify the starting offset and the following processes would then have offset 0 by default. Though we can discuss more about your logic if it is less complicated

**Pull request merged**